newIDE/app/src/ObjectsRendering/PixiResourcesLoader.js

```diff
@@ -146,6 +147,133 @@ export default class PixiResourcesLoader {
     return loadedTextures[resourceName];
   }
 
+  static createTileSetResource(
+    tiledData: any,
+    tex: any,
+    requestedTileSetId: string,
+    onLoad: any => void
+  ) {
+    // Todo implement tileset index and use it instead of 0
+    const { tilewidth, tilecount, tileheight, tiles } = tiledData.tilesets[0];
+    console.log('NEW TILESET data::', tilewidth, tilecount, tileheight, tiles);
+    const textureCache = new Array(tilecount).fill(0).map((_, frame) => {
+      const cols = Math.floor(tex.width / tilewidth);
+      const x = ((frame - 1) % cols) * tilewidth;
+      const y = Math.floor((frame - 1) / cols) * tileheight;
+      const rect = new PIXI.Rectangle(x, y, tilewidth, tileheight);
+      const texture = new PIXI.Texture(tex, rect);
+      texture.baseTexture.scaleMode = PIXI.SCALE_MODES.NEAREST;
+      texture.cacheAsBitmap = true;
+
+      return texture;
+    });
+    const newTileset = {
+      width: tex.width,
+      height: tex.height,
+      tilewidth,
+      tileheight,
+      texture: tex,
+      textureCache,
+      layers: tiledData.layers,
+      tiles,
+      tilecount,
+    };
+    onLoad(newTileset);
+    loadedTileSets[requestedTileSetId] = newTileset;
+  }
+
+  // If a Tileset changes (json or image), a tilemap using it needs to re-render
+  // The tileset needs to be rebuilt for use
+  // But we need to have the capacity to instance a tilemap, so more than one tilemaps with different tileset layers
+  // We need to cache the tileset somewhere, the tilemap instance will have to re-render it
+  // Tileset changes => rerender it => tilemaps using it rerender too (keeping their layer visibility option)
+  // tileset id == jsonResourceName + imageResourcename
+  static getPIXITileSet(
```

Comment:
Same for the other methods, I wonder if we could move them to JsExtension.js

That's no big deal, but for the Runtime if you're coding the equivalent of these/copy pasting them, I suggest to **not** put them inside `gdjs.PixiImageManager`, instead create a standalone class to avoid "polluting" `gdjs.PixiImageManager`.

Reply:
> Same for the other methods, I wonder if we could move them to JsExtension.js
> 
> That's no big deal, but for the Runtime if you're coding the equivalent of these/copy pasting them, I suggest to **not** put them inside `gdjs.PixiImageManager`, instead create a standalone class to avoid "polluting" `gdjs.PixiImageManager`.

gdjs.PixiTilesetManager ? :)

Reply:
Yep sounds good! :)